src/handlers/extract/handler.ts

```diff
@@ -1,12 +1,47 @@
-export const handler = async (): Promise<string | undefined> => {
-  if (
-    String(process.env.EXTRACTION_HANDLER_THROW_ERROR).toLowerCase() === "true"
-  ) {
-    const message =
-      "Environment variable EXTRACTION_HANDLER_THROW_ERROR is true";
-    console.error(message);
-    throw new Error(message);
-  }
-
-  return process.env.EXTRACTION_HANDLER_RETURN_VALUE;
+import * as AWS from "aws-sdk";
+import { S3Event } from "aws-lambda";
+
+export const handler = async (
+  event: S3Event
+): Promise<AWS.Textract.StartExpenseAnalysisResponse[]> => {
+  // Set up
+  const textract = new AWS.Textract({ region: "eu-west-2" });
+  const response: AWS.Textract.StartExpenseAnalysisResponse[] = [];
+
+  // Get Bucket and filename from the event.
+  const promises = event.Records.map(async (record) => {
+    console.log(`Record ${JSON.stringify(record)}`);
+    const bucket = record.s3.bucket.name;
+    const fileName = record.s3.object.key;
+    console.log("File Name: " + fileName);
+
+    if (process.env.TEXT_EXTRACT_ROLE === undefined) {
```

Comment:
You should check this and the SNS-topic config first thing in the function as you will not be able to continue without it anyway.